renderer/brave_content_settings_observer.cc

```diff
@@ -243,52 +243,36 @@ bool BraveContentSettingsObserver::AllowAutoplay(bool default_value) {
     }
   }
 
-  // in the absence of an explicit block rule, whitelist the following sites
-  std::string kWhitelistPatterns[] = {
-      "[*.]example.com",
-      "[*.]youtube.com",
-      "[*.]khanacademy.org",
-      "[*.]twitch.tv",
-      "[*.]vimeo.com",
-      "[*.]udemy.com",
-      "[*.]duolingo.com",
-      "[*.]giphy.com",
-      "[*.]imgur.com",
-      "[*.]netflix.com",
-      "[*.]hulu.com",
-      "[*.]primevideo.com",
-      "[*.]dailymotion.com",
-      "[*.]tv.com",
-      "[*.]viewster.com",
-      "[*.]metacafe.com",
-      "[*.]d.tube",
-      "[*.]spotify.com",
-      "[*.]lynda.com",
-      "[*.]soundcloud.com",
-      "[*.]pandora.com",
-      "[*.]periscope.tv",
-      "[*.]pscp.tv",
-      "[*.]hangouts.google.com",
-      "[*.]meet.google.com",
-      "[*.]rainway.com",
-      "[*.]rainway.io",
-      "[*.]cheddar.com",
-  };
-  for (const auto& pattern : kWhitelistPatterns) {
-    if (ContentSettingsPattern::FromString(pattern).Matches(primary_url))
-      return true;
-  }
-
   blink::mojom::blink::PermissionServicePtr permission_service;
 
   render_frame()->GetRemoteInterfaces()
     ->GetInterface(mojo::MakeRequest(&permission_service));
 
   if (permission_service.get()) {
-    auto descriptor = blink::mojom::blink::PermissionDescriptor::New();
-    descriptor->name = blink::mojom::blink::PermissionName::AUTOPLAY;
-    permission_service->RequestPermission(std::move(descriptor), true,
-                                          base::DoNothing());
+    // Check (synchronously) whether we already have permission to autoplay.
```

Comment:
I guess there's no way to avoid going synchronous here, as we need to return a Boolean from this function?

Reply:
Correct.